packages/toc/src/generators/notebook/options_manager.ts

```diff
@@ -176,6 +173,21 @@ class OptionsManager extends Registry.IOptionsManager {
     this._widget.update();
   }
 
+  /**
+   * Updates a table of contents widget and
+   * emits a signal in case an extension wants
+   * to perform an action when the collapse button
+   * is pressed.
+   */
+  updateAndCollapse(heading: IHeading, state: boolean) {
```

Comment:
Are these likely to change over time, i.e. should this be a typed object that is passed?

Reply:
I changed it to taking `ICollapseChangedArgs`, do you think that's a good solution?